driver-core/src/test/unit/com/mongodb/operation/AsyncQueryBatchCursorSpecification.groovy

```diff
@@ -453,33 +453,25 @@ class AsyncQueryBatchCursorSpecification extends Specification {
                 cursor.close()
                 it[6].onResult(response, null)
             }
-            1 * connectionB.commandAsync(NAMESPACE.databaseName, createKillCursorsDocument(initialResult.cursor), _, primary(),
-                    _, _, _) >> {
-                it[6].onResult(null, null)
-            }
         } else {
             1 * connectionA.getMoreAsync(_, _, _, _, _) >> {
                 // Simulate the user calling close while the getMore is in flight
                 cursor.close()
                 it[4].onResult(response, null)
             }
-            1 * connectionB.killCursorAsync(NAMESPACE, [initialResult.cursor.id], _) >> {
-                it[2].onResult(null, null)
-            }
         }
 
         then:
-        thrown(MongoException)
+        noExceptionThrown()
```

Comment:
I don't follow this... What was the change that makes this not throw anymore?

Reply:
We now have the `isClosePending` flag so that we do not prematurely close the cursor while an operation is in progress.